### PR TITLE
Support Route Scope Binding

### PR DESCRIPTION
**Pull request by ahmedbally**
This PR will help to support route scope binding

### CODE REVIEW

**Comment by antonkomarev** [comment]
@ahmedbally thank you for the contribution! Will review it asap.

File: src/Traits/OptimusEncodedRouteKey.php [hunk 1]
```diff
@@ -40,6 +40,31 @@ public function resolveRouteBinding($value, $field = null)
...
+
+        if (is_string($value) && ctype_digit($value)) {
+            $value = (int) $value;
+        }
+
+        if (is_int($value) && $field === $this->getRouteKeyName()) {
+            $value = $this->getOptimus()->decode($value);
+        }
+
+        return $query->where($field ?? $this->getRouteKeyName(), $value);
```
Comment (antonkomarev) [inline src/Traits/OptimusEncodedRouteKey.php:65]: `$field` cannot be `null`, you are setting it on the line 54.

And if we remove null coalescence, then this method will be identical to the `resolveRouteBinding`.

Comment (ahmedbally) [reply src/Traits/OptimusEncodedRouteKey.php]: sorry fixed

**Comment by ahmedbally** [comment]
@antonkomarev  any update

**Comment by antonkomarev** [comment]
Published [Laravel Optimus v3.10.0](https://github.com/cybercog/laravel-optimus/releases/tag/3.10.0) with your changes.

**Pull request merged**